rosbag2_py/src/rosbag2_py/_writer.cpp

```diff
@@ -105,28 +85,38 @@ std::unordered_set<std::string> get_registered_serializers()
 
 }  // namespace rosbag2_py
 
+using PyWriter = rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>;
+using PyCompressionWriter = rosbag2_py::Writer<rosbag2_compression::SequentialCompressionWriter>;
+
 PYBIND11_MODULE(_writer, m) {
   m.doc() = "Python wrapper of the rosbag2_cpp writer API";
 
-  pybind11::class_<rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>>(
-    m, "SequentialWriter")
-  .def(pybind11::init())
-  .def("open", &rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>::open)
-  .def("write", &rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>::write)
-  .def("remove_topic", &rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>::remove_topic)
-  .def("create_topic", &rosbag2_py::Writer<rosbag2_cpp::writers::SequentialWriter>::create_topic);
-
-  pybind11::class_<rosbag2_py::Writer<rosbag2_compression::SequentialCompressionWriter>>(
-    m, "SequentialCompressionWriter")
+  pybind11::class_<PyWriter>(m, "SequentialWriter")
   .def(pybind11::init())
-  .def("open", &rosbag2_py::Writer<rosbag2_compression::SequentialCompressionWriter>::open)
-  .def("write", &rosbag2_py::Writer<rosbag2_compression::SequentialCompressionWriter>::write)
   .def(
-    "remove_topic",
-    &rosbag2_py::Writer<rosbag2_compression::SequentialCompressionWriter>::remove_topic)
+    "open",
+    pybind11::overload_cast<
+      const rosbag2_storage::StorageOptions &, const rosbag2_cpp::ConverterOptions &
+    >(&PyWriter::open))
+  .def("write", &PyWriter::write)
+  .def("close", &PyWriter::close)
+  .def("remove_topic", &PyWriter::remove_topic)
+  .def("create_topic", &PyWriter::create_topic)
+  .def("take_snapshot", &PyWriter::take_snapshot)
+  ;
```

Comment:
@JannikJose Out of curiosity why `.def("split_bagfile", &PyWriter::split_bagfile)` missed from the original PR https://github.com/ros2/rosbag2/pull/1220/files#diff-f3aa422fd35406262b9209321476913d230c5bcd81b395525719e0166395ce90R103 ?

Reply:
@MichaelOrlov The reason I left out the `split_bagfile` definition is that it doesn't seem to be implemented in the [Humble C++ Writer](https://github.com/ros2/rosbag2/blob/humble/rosbag2_cpp/include/rosbag2_cpp/writer.hpp) yet. Because of that the build fails when adding the `split_bagfile` definition.
I also found the PR #1124 which rejected the Humble backport because it would have broken the ABI.

It seems to be implemented in the `SequentialWriter` though: https://github.com/ros2/rosbag2/blob/e12f652720728211fe53684334b977cd7ecd1f27/rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp#L327C1-L330C2